### PR TITLE
Set data into redux store

### PR DESCRIPTION
**Pull request by anzhelikavelychko**
Set up redux, and store user data from the Sign In/ Sign out form
 was made the file Protected route

### CODE REVIEW

**Review by romanolytvyn — APPROVED** [review]

**Review by rostyslav-vasyliuk — APPROVED** [review]

**Review by machikharostyk — APPROVED** [review]

**Review by OBats — APPROVED** [review]

File: src/Routes.jsx [hunk 1]
```diff
@@ -42,13 +30,17 @@ class Routes extends Component {
...
           <Redirect exact to="/" />
         </Switch>
       </React.Fragment>
     );
   }
 }
 
-export default Routes;
+const mapStateToProps = ({ user }) => ({
+  userData: user.userData,
```
Comment (Antoniozinchenko) [inline src/Routes.jsx]: connect userData directly to NavBar, not in this component

File: src/api/auth-api.js [hunk 2]
```diff
@@ -1,17 +1,15 @@
 import http from './http';
 
 export const signIn = async (values) => {
-  await http.post('/api/auth/signin', values)
-    .then((req) => {
-      localStorage.setItem('access-token', req.data);
-    });
+  const response = await http.post('/api/auth/signin', values);
+  localStorage.setItem('access-token', response.data);
```
Comment (Antoniozinchenko) [inline src/api/auth-api.js]: set the same as you write for signup
image

File: package.json [hunk 3]
```diff
@@ -12,9 +12,14 @@
     "react-dom": "^16.7.0",
     "react-markdown": "^4.0.6",
     "react-places-autocomplete": "^7.2.0",
+    "react-redux": "^6.0.0",
     "react-router-dom": "^4.3.1",
     "react-scripts": "2.1.3",
     "react-select": "^2.3.0",
+    "redux": "^4.0.1",
+    "redux-devtools-extension": "^2.13.8",
+    "redux-logger": "^3.0.6",
```
Comment (Antoniozinchenko) [inline package.json]: not used module, should be removed

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

**Review by OBats — APPROVED** [review]

File: src/components/auth/SignOut.jsx [hunk 4]
```diff
@@ -11,4 +14,10 @@ class Logout extends Component {
   }
 }
 
-export default Logout;
+const mapDispatchToProps = dispatch => ({
+  handleLogOut: () => {
+    dispatch(handleLogOut());
```
Comment (Antoniozinchenko) [inline src/components/auth/SignOut.jsx]: this action should be dispatched on button click, not on component did mount. remove this component at all

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

File: src/Routes.jsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/prefer-stateless-function */
```
Comment (Antoniozinchenko) [inline src/Routes.jsx]: of course  it should be stateless! write Routes as simple functional component

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

**Review by Antoniozinchenko — APPROVED** [review]

**Pull request merged**